File: app/src/main/java/com/pistis/vdoc/activities/SintomaActivity.java
package com.pistis.vdoc.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.pistis.vdoc.database.DBSPopulater;
import com.pistis.vdoc.database.DataBaseStorage;
import com.pistis.vdoc.entities.Doenca;

import java.util.ArrayList;

import symptomsme.symptomsme.empsoft.projeto.symptomsme.R;

public class SintomaActivity extends AppCompatActivity {

    private DataBaseStorage mDb;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

       // requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_sintoma);

        mDb = new DataBaseStorage(getApplicationContext());

        if(mDb.getAllDoencas().size() == 0){
            DBSPopulater populater = new DBSPopulater(getApplicationContext());
            populater.populateBD();
        }
        final EditText etSintomas = (EditText) findViewById(R.id.et_sintoma);

        ImageView btPesquisar = (ImageView) findViewById(R.id.bt_pesquisar);

        btPesquisar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if(etSintomas.getText().toString().trim().equals("")){
                    Toast.makeText(getApplicationContext(),"Type any symptom",Toast.LENGTH_LONG).show();
                }

                else{

                    String[] sintomas = (etSintomas.getText().toString().trim()).split(",");
                    ArrayList<Doenca> doencas = mDb.getDoencasBySintomas(sintomas);

                    if(doencas.size() == 0)
                        Toast.makeText(getApplicationContext(),"No disease found",Toast.LENGTH_LONG).show();
                    else{
                        Intent i = new Intent(getApplicationContext(), DoencasActivity.class);
                        i.putExtra("doencas",doencas);

                        startActivity(i);
                    }
                }
            }
        });

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/pistis/vdoc/activities/SintomaActivity.kt
package com.pistis.vdoc.activities

import symptomsme.symptomsme.empsoft.projeto.symptomsme.R
import com.pistis.vdoc.database.DataBaseStorage
import com.pistis.vdoc.database.DBSPopulater
import android.os.Bundle
import android.content.Intent
import android.view.*
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class SintomaActivity : AppCompatActivity() {
    private var mDb: DataBaseStorage? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_sintoma)

        mDb = DataBaseStorage(applicationContext)
        if (mDb!!.getAllDoencas()?.size == 0) {
            val populater = DBSPopulater(applicationContext)
            DBSPopulater.Companion.populateBD()
        }
        val etSintomas = findViewById<View?>(R.id.et_sintoma) as EditText
        val btPesquisar = findViewById<View?>(R.id.bt_pesquisar) as ImageView
        btPesquisar.setOnClickListener {
            if (etSintomas.text.toString().trim { it <= ' ' } == "") {
                Toast.makeText(applicationContext, "Type any symptom", Toast.LENGTH_LONG).show()
            } else {
                val sintomas: Array<String?> = etSintomas.text.toString().trim { it <= ' ' }
                    .split(",").toTypedArray()
                val doencas = mDb!!.getDoencasBySintomas(sintomas)
                if (doencas != null) {
                    if (doencas.size == 0) Toast.makeText(
                        applicationContext,
                        "No disease found",
                        Toast.LENGTH_LONG
                    ).show() else {
                        val i = Intent(applicationContext, DoencasActivity::class.java)
                        i.putExtra("doencas", doencas)
                        startActivity(i)
                    }
                }
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }
}